### PR TITLE
lang: Add support of user custom languague.

### PR DESCRIPTION
**Pull request by Zangetsu38**
# About
- Implement featur request by https://github.com/Vita3K/Vita3K/issues/3071

### CODE REVIEW

**Comment by Macdu** [comment]
I'm not sure this approach (making a system and user directory inside lang) will work on linux as the static assets directory can be different compared to the shared folder.

**Comment by Zangetsu38** [comment]
I will admit to you that I haven't thought about Linux, but in itself there shouldn't be a problem, it's inside the lang folder, so whether there is another folder or not, it normally doesn't change anything , like the data folder

**Comment by Macdu** [comment]
I agree that this still works. But in this case creating in the data folder the file "user/PLACE USER LANG HERE.txt" does not make a lot of sense. It would be better in some init function to create the file shared_path / "data/user/PLACE USER LANG HERE.txt" in some init function if does not already exists.

**Comment by Zangetsu38** [comment]
hmm I see, on the other hand I want to leave that in the lang folder, it doesn't make sense otherwise.
The goal is for users to upload and test their translated language file, and then for it to be directly added to the project here via a pr.

so this txt is just here for indicate to user where place lang file

**Comment by Macdu** [comment]
yes, so if you create the shared_path / "data/lang/user/PLACE USER LANG HERE.txt" in some init function, on windows it will be in the right place (in the lang folder) while on Linux, it will be in its correct read-write location.

**Comment by Zangetsu38** [comment]
> yes, so if you create the shared_path / "data/lang/user/PLACE USER LANG HERE.txt" in some init function, on windows it will be in the right place (in the lang folder) while on Linux, it will be in its correct read-write location.

I have said i want it inside lang, but, here you said shared_path/data/lang, and me i wxant so base_path/lang, no data

**Comment by Zangetsu38** [comment]
> yes, so if you create the shared_path / "data/lang/user/PLACE USER LANG HERE.txt" in some init function, on windows it will be in the right place (in the lang folder) while on Linux, it will be in its correct read-write location.

I'm not sure I understand, I'm talking to you about the lang folder at the root of the base_path, but here you're telling me about the data folder, which is a read-only folder by the way

On the Windows side it must be base_path/lang/user
There is no question of data folder here

**Comment by Macdu** [comment]
Ah yes sorry, remove the data part, for some reason I thought lang was in a data folder

**Comment by Zangetsu38** [comment]
> Ah yes sorry, remove the data part, for some reason I thought lang was in a data folder

I was also saying, you lost me there lol, i have coded lang in root so lol, but so the shared path is good for writing on Linux? this all new path lost me0..

**Comment by Macdu** [comment]
Yes, basically, basic_path and static_path can be read-only while shared-path should be guaranteed to be read-write (at least that's how I understood it). On windows, these 3 paths are the same.

**Comment by Zangetsu38** [comment]
i see hmm

**Comment by Zangetsu38** [comment]
is normally done like you request @Macdu

File: vita3k/lang/src/lang.cpp [hunk 1]
```diff
@@ -67,10 +67,19 @@ void init_lang(LangState &lang, EmuEnvState &emuenv) {
...
     }
 
-    pugi::xml_document lang_xml;
+    const auto system_lang_path{ emuenv.static_assets_path / "lang/system" };
+    const auto user_lang_path{ emuenv.shared_path / "lang/user" };
+
+    // Create user lang folder if not exists and create a file to indicate where to place the lang files
+    if (!fs::exists(user_lang_path)) {
+        fs::create_directories(user_lang_path);
+        std::ofstream outfile((user_lang_path / "PLACE USER LANG HERE.txt").string());
```
Comment (Macdu) [inline vita3k/lang/src/lang.cpp:76]:
```suggestion
        fs::ofstream outfile(user_lang_path / "PLACE USER LANG HERE.txt");
```

Comment (Zangetsu38) [reply vita3k/lang/src/lang.cpp]: done

**Review by Macdu — APPROVED** [review]

**Pull request merged**